lib/ansible/plugins/connection/psrp.py

```diff
@@ -457,14 +461,29 @@ def exec_command(self, cmd: str, in_data: bytes | None = None, sudoable: bool =
                 display.vvv("PSRP: EXEC (via pipeline wrapper)")
             else:
                 display.vvv("PSRP: EXEC %s" % script, host=self._psrp_host)
+
+        elif cmd.startswith(f"{common_args_prefix} -File "):  # trailing space is on purpose
+            # Used when executing a script file, we will execute it in the runspace process
+            # instead on a new subprocess
+            script = 'param([string]$Path, [Parameter(ValueFromRemainingArguments)][string[]]$ScriptArgs) & $Path @ScriptArgs'
+
+            # Using shlex isn't perfect but it's good enough.
+            cmd = cmd[len(common_args_prefix) + 7:]
+            script_args = shlex.split(cmd)
```

Comment:
I'm amazed this hasn't nailed us somehow yet, but not a lot better we can do without an exec subsystem

Reply:
Yea considering we haven't had any complaints I didn't think it was worth the extra effort to solve. IIRC this was only for modules being invoked in a non-pipeline manner which is probably 0.1% of the use cases in Ansible Windows.